Assets/Scripts/SlownessOnImpact.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SlownessOnImpact : MonoBehaviour {
+
+	public float slownessFactor = 10;
+	void Start () {
+		
+	}
+	
+	// Update is called once per frame
+	void Update () {
+	}
+
+	private void OnTriggerEnter2D(Collider2D e)
+	{
+		if(e.gameObject.GetComponent<PathMovement> ()){
+			e.gameObject.GetComponent<PathMovement> ().speed = e.gameObject.GetComponent<PathMovement>().speed / slownessFactor;
+		}
+	}
+
+	private void OnTriggerExit2D(Collider2D e)
+	{
+		if(e.gameObject.GetComponent<PathMovement> ()){
+			e.gameObject.GetComponent<PathMovement> ().speed = e.gameObject.GetComponent<PathMovement>().speed * slownessFactor;
```

Comment:
Need to tie to speed reset method, as this will cause speed boost perminatly if speed effecting pickup is collected abd speed is reset in black hole (part of making effect stackable)

use: e.gameObject.SendMessage("SetSpeedRevertTimer", 0);

let the speed reverter handle reversion, as it causes issues if not

note that this will cause an issue if there is time left on the duration of an effect.